v2/api/apimanagement/v1api20220801/structure.txt

```diff
@@ -449,103 +674,193 @@ github.com/Azure/azure-service-operator/v2/api/apimanagement/v1api20220801
-    │   │   └── StoreName: *Enum (2 values)
-    │   │       ├── "CertificateAuthority"
-    │   │       └── "Root"
-    │   ├── CustomProperties: map[string]string
-    │   ├── DisableGateway: *bool
-    │   ├── EnableClientCertificate: *bool
-    │   ├── HostnameConfigurations: Object (11 properties)[]
-    │   │   ├── Certificate: *Object (3 properties)
-    │   │   │   ├── Expiry: *string
-    │   │   │   ├── Subject: *string
-    │   │   │   └── Thumbprint: *string
-    │   │   ├── CertificatePassword: *string
-    │   │   ├── CertificateSource: *Enum (4 values)
-    │   │   │   ├── "BuiltIn"
-    │   │   │   ├── "Custom"
-    │   │   │   ├── "KeyVault"
-    │   │   │   └── "Managed"
-    │   │   ├── CertificateStatus: *Enum (3 values)
-    │   │   │   ├── "Completed"
-    │   │   │   ├── "Failed"
-    │   │   │   └── "InProgress"
-    │   │   ├── DefaultSslBinding: *bool
-    │   │   ├── EncodedCertificate: *string
-    │   │   ├── HostName: *string
-    │   │   ├── IdentityClientId: *string
-    │   │   ├── KeyVaultId: *string
-    │   │   ├── NegotiateClientCertificate: *bool
-    │   │   └── Type: *Enum (5 values)
-    │   │       ├── "DeveloperPortal"
-    │   │       ├── "Management"
-    │   │       ├── "Portal"
-    │   │       ├── "Proxy"
-    │   │       └── "Scm"
-    │   ├── NatGatewayState: *Enum (2 values)
-    │   │   ├── "Disabled"
-    │   │   └── "Enabled"
-    │   ├── NotificationSenderEmail: *string
-    │   ├── PublicIpAddressId: *string
-    │   ├── PublicNetworkAccess: *Enum (2 values)
-    │   │   ├── "Disabled"
-    │   │   └── "Enabled"
-    │   ├── PublisherEmail: *string
-    │   ├── PublisherName: *string
-    │   ├── Restore: *bool
-    │   ├── VirtualNetworkConfiguration: *Object (1 property)
-    │   │   └── SubnetResourceId: *string
-    │   └── VirtualNetworkType: *Enum (3 values)
-    │       ├── "External"
-    │       ├── "Internal"
-    │       └── "None"
-    ├── Sku: *Object (2 properties)
-    │   ├── Capacity: *int
-    │   └── Name: *Enum (6 values)
-    │       ├── "Basic"
-    │       ├── "Consumption"
-    │       ├── "Developer"
-    │       ├── "Isolated"
-    │       ├── "Premium"
-    │       └── "Standard"
-    ├── Tags: map[string]string
-    └── Zones: string[]
+├── Service_Spec_ARM: Object (7 properties)
+│   ├── Identity: *Object (2 properties)
+│   │   ├── Type: *Enum (4 values)
+│   │   │   ├── "None"
+│   │   │   ├── "SystemAssigned"
+│   │   │   ├── "SystemAssigned, UserAssigned"
+│   │   │   └── "UserAssigned"
+│   │   └── UserAssignedIdentities: map[string]Object (0 properties)
+│   ├── Location: *string
+│   ├── Name: string
+│   ├── Properties: *Object (16 properties)
+│   │   ├── AdditionalLocations: Object (7 properties)[]
+│   │   │   ├── DisableGateway: *bool
+│   │   │   ├── Location: *string
+│   │   │   ├── NatGatewayState: *Enum (2 values)
+│   │   │   │   ├── "Disabled"
+│   │   │   │   └── "Enabled"
+│   │   │   ├── PublicIpAddressId: *string
```

Comment:
This one is in the _STATUS object - should I do this?

```yaml
  - name: AdditionalLocation_STATUS
    group: apimanagement
    property: PublicIpAddressId
    remove: true
    because: This property should be marked readonly
```

Reply:
No status ones should be left as-is. They display the real ARM ID.

Reply:
Oh, this is also the ARM type so it doesn't apply here.

Reply:
Resolving as _ARM properties should be strings